components/availability/src/Availability.svelte

```diff
@@ -872,44 +874,47 @@
     }
 
     if (Array.isArray(calendarsToFetch) && calendarsToFetch.length > 0) {
-      // Note: we split our requests by email_id as calendars/availability doesn't support partial availability amongst a group
-      return Promise.all(
-        calendarsToFetch.map(async (user) => {
-          const { email, token } = user;
-          let availableTimeslots = await $AvailabilityStore[
-            JSON.stringify({
-              ...getAvailabilityQuery([email], token ?? access_token),
-              forceReload,
-            })
-          ];
-
-          // Blob consecutive timeslots together for easier overlap logic elsewhere
-          let groupedSlots = groupConsecutiveTimeslots(
-            availableTimeslots.time_slots,
-          );
-          return {
-            email,
-            time_slots: groupedSlots,
-          };
-        }),
-      ).then(async (consolidatedAvailabilityForGivenDay) => {
-        loading = false;
-        for (const user of consolidatedAvailabilityForGivenDay) {
-          freeBusyCalendars.push({
-            emailAddress: user.email,
-            account: {
-              emailAddress: user.email, // ¯\_(ツ)_/¯
-            },
-            availability: AvailabilityStatus.FREE,
-            timeslots: user.time_slots.map((slot: PreDatedTimeSlot) => ({
+      const fetchedCalendars = await $AvailabilityStore[
+        JSON.stringify({
+          ...getAvailabilityQuery(
+            calendarsToFetch.map((x) => x.email),
+            access_token,
+          ),
+          forceReload,
+        })
+      ];
+
+      loading = false;
+
+      const timeSlotMap: Record<string, PreDatedTimeSlot[]> = {};
+
+      for (const user of fetchedCalendars?.order) {
+        timeSlotMap[user] = [];
+      }
+
+      for (const slot of fetchedCalendars.time_slots) {
+        slot.emails.forEach((e) => timeSlotMap[e].push(slot));
```

Comment:
would it be better to combine these? 
```suggestion
      for (const slot of fetchedCalendars.time_slots) {
        slot.emails.forEach((e) => 
          (Array.isArray(timeSlotMap[e]) ? 
            timeSlotMap[e].push(slot) :
            timeSlotMap[e] = [slot]);
```

I think this reduces performance to `n` from `n * m`

Even if it's reasonable to assume that `n * m` won't be a lot most cases, I think it's still better to make this change for cases like ShareCase